src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -125,14 +131,16 @@ public boolean link(ProcessPaths paths, ProjectConfiguration projectConfiguratio
             throw new IllegalArgumentException("Linking failed, since there is no objectfile named "+objectFilename+" under "
                     +gvmPath.toString());
         }
-        ProcessBuilder linkBuilder = new ProcessBuilder("gcc");
+        ProcessBuilder linkBuilder = new ProcessBuilder(getLinker());
         Path appPath = gvmPath.resolve(appName);
 
         linkBuilder.command().add("-o");
         linkBuilder.command().add(paths.getAppPath().resolve(appName).toString());
 
-        Arrays.stream(C_RESOURCES)
-              .forEach( r -> linkBuilder.command().add(appPath.resolve(r + ".o").toString()));
+        getAdditionalSourceFiles()
+              .forEach( r -> linkBuilder.command().add(
+                      appPath.resolve(r.replaceAll("\\..*", ".o")).toString()));
```

Comment:
Non problem with that change, but IMO adding different extension is much simpler then regex based replace :)

Reply:
I'm ok with that, but if you just say "foo" how do we know we need to search for foo.c or foo.m or foo.h?

Reply:
The way I thought was that we have the file names but add extentions when we need. So in in place we added ".c" to them, in another we added ".o". We looping through all the files anyway. But... either way would work.